gnc/sub8_perception/nodes/buoy_2d.py

```diff
@@ -165,30 +201,24 @@ def find_single_buoy(self, img, buoy_type):
         contour, tuple_center, area = best_ret
         true_center, rad = cv2.minEnclosingCircle(contour)
 
-        if self.camera_model is not None:
+        if self.camera_model is not None and (buoy_type == 'red'):
             self.rviz.draw_ray_3d(tuple_center, self.camera_model, self.draw_colors[buoy_type])
-
-            if buoy_type == 'red' and not self.done_once:
-                # self.done_once = True
-
-                (t, rot_q) = self.transformer.lookupTransform('/map', '/stereo_front/right', self.last_image_time)
-                trans = np.array(t)
-
-                if (self.last_t is None) or (np.linalg.norm(trans - self.last_t) > 0.3):
-                    self.last_t = trans
-                    self.ppf.set_pose(trans, sub8_ros_tools.geometry_helpers.quaternion_matrix(rot_q))
-                    est, cov = self.ppf.observe(true_center)
-                    self.rviz.draw_sphere(est, color=(0.9, 0.1, 0.0, 1.0), scaling=tuple(cov), frame='/map')
-                    k = 6
-                    for particle in self.ppf.particles[:, ::50].transpose():
-                        k += 1
-                        self.rviz.draw_sphere(
-                            particle,
-                            color=(0.8, 0.2, 0.0, 0.7),
-                            scaling=(0.05, 0.05, 0.05),
-                            _id=k,
-                            frame='/map'
-                        )
+            (t, rot_q) = self.transformer.lookupTransform('/map', '/stereo_front/right', self.last_image_time - rospy.Duration(0.04))
```

Comment:
Is this to fix the camera stamped in the future problem? Regardless @rleyva should know about it if you haven't told him.